### DISCUSSION TITLE
magit-branch-read-upstream-first nil should check branch name doesn't already exist before asking for starting branch

### DISCUSSION THREAD

**Discussion by verhovsky** [magit/magit#4837] [Ideas]
To reproduce: 

set this in your config

```
(setq magit-branch-read-upstream-first nil)
```

then create a new branch, it will ask you for the name of the new branch first, type in the name of an already existing branch and press Enter.

Expected:

I immediately see the "a branch named "foo" already exists" error.

Actual:

I am asked to type/choose the "Create and checkout branch `foo` at:" dialog, I have to choose the starting branch even though we can already know that the operation is going to fail, regardless of what I choose.

**Comment by tarsius** [comment]
Thanks for the suggestion! I've implemented it in cd6fbe28873c1ec973bfe51af0f3ade5d69e9ba6.
